### PR TITLE
Implement PeerDAS RPC handlers

### PR DESCRIPTION
**Pull request by dapplion**
## Issue Addressed

Continue porting network changes of `das` branch. Extends https://github.com/sigp/lighthouse/pull/6238

Part of 
- https://github.com/sigp/lighthouse/issues/4983

_all branches are on the sipg/lighthouse repo to start another PR conga :)_

## Proposed Changes

- Implement serving RPC handlers for `data_columns_by_root` and `data_columns_by_range`. 
- This PR **excludes** the feature to reprocess early sampling requests.

Upstream PRs:

- https://github.com/sigp/lighthouse/pull/5196

### CODE REVIEW

**Comment by jimmygchen** [comment]
Oops I thought this was the first PR in the conga line and incorrectly set the base branch to `unstable`. I've reverted my merge commit and it's now back to what it was before.

**Comment by jimmygchen** [comment]
I've merged latest base branch into this branch so i can extend the conga line.

File: beacon_node/network/src/network_beacon_processor/rpc_methods.rs [hunk 1]
```diff
@@ -320,14 +320,53 @@ impl<T: BeaconChainTypes> NetworkBeaconProcessor<T> {
     pub fn handle_data_columns_by_root_request(
```
Comment (realbigsean) [inline beacon_node/network/src/network_beacon_processor/rpc_methods.rs:320]: should this use `terminate_response_stream` like we use in other similar methods?

File: beacon_node/network/src/network_beacon_processor/rpc_methods.rs [hunk 2]
```diff
@@ -320,14 +320,53 @@ impl<T: BeaconChainTypes> NetworkBeaconProcessor<T> {
...
+                    return;
+                }
+            }
+        }
+
+        debug!(
+            self.log,
+            "Received DataColumnsByRoot Request";
+            "peer" => %peer_id,
+            "request" => ?request.group_by_ordered_block_root(),
```
Comment (realbigsean) [inline beacon_node/network/src/network_beacon_processor/rpc_methods.rs:376]: what is the benefit of ordering by block roots in this log?

**Review by realbigsean — APPROVED** [review]
one small suggestion and one question but looks good

File: beacon_node/network/src/network_beacon_processor/rpc_methods.rs (hunk 2 above)
Comment (dapplion) [reply beacon_node/network/src/network_beacon_processor/rpc_methods.rs:376]: I introduced `group_by_ordered_block_root` originally to handle the reprocessing of requests. I left it here as it produces a more succint output. The raw request repeats the root for every column. Usually peers requests multiple indexes for the same root, so the output of this is nicer.

File: beacon_node/network/src/network_beacon_processor/rpc_methods.rs (hunk 1 above)
Comment (dapplion) [reply beacon_node/network/src/network_beacon_processor/rpc_methods.rs:320]: Added in aa0b9037c689de617a9092cdfc0bc9fa5283a4f2

**Comment by dapplion** [comment]
@jimmygchen I have forced push a rebase of this branch against unstable, as the commit from https://github.com/sigp/lighthouse/pull/6238 had leaked into this PR

**Comment by realbigsean** [comment]
@mergify queue

**Comment by realbigsean** [comment]
@mergify refresh

**Comment by realbigsean** [comment]
@mergify queue

**Comment by realbigsean** [comment]
@mergify dequeue

**Comment by realbigsean** [comment]
@mergify queue

**Comment by realbigsean** [comment]
@mergify requeue

**Pull request merged**